connection_producer.go

```diff
@@ -2,28 +2,24 @@ package redis
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
 	"fmt"
-	"strings"
 	"sync"
 
-	"github.com/hashicorp/errwrap"
 	"github.com/hashicorp/vault/sdk/database/helper/connutil"
 	"github.com/mediocregopher/radix/v4"
 	"github.com/mitchellh/mapstructure"
 )
 
 type redisDBConnectionProducer struct {
-	PublicKey   string `json:"public_key"`
-	PrivateKey  string `json:"private_key"`
-	ProjectID   string `json:"project_id"`
-	Host        string `json:"host"`
-	Port        int    `json:"port"`
-	Username    string `json:"username"`
-	Password    string `json:"password"`
-	TLS         bool   `json:"tls"`
-	InsecureTLS bool   `json:"insecure_tls"`
-	Base64Pem   string `json:"base64pem"`
-	BucketName  string `json:"bucket_name"`
+	Host     string `json:"host"`
+	Port     int    `json:"port"`
+	Username string `json:"username"`
+	Password string `json:"password"`
+	TLS      bool   `json:"tls"`
+	CaCrt    string `json:"cacrt"`
```

Comment:
We should add https://www.vaultproject.io/api-docs/secret/databases/couchbase#insecure_tls similar to couchbase. Will set it below in `InsecureSkipVerify`.